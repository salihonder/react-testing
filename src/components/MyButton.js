const MyButton = (props) => <button onClick={props.alert}>Some</button>

export default MyButton;
