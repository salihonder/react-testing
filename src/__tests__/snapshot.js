import { render } from '@testing-library/react';

// Components
import MyButton from '../components/MyButton';

test("Snapshot test for Header", () => {
    // ARRANGE
    const Header = () => <h1 style={{ color: 'red' }}>Something</h1>

    // ACT
    const { asFragment } = render(<Header />);

    // ASSERT
    expect(asFragment()).toMatchSnapshot();
})

test("Snapshot test for Button", () => {
    // ARRANGE
    const alertFunction = jest.fn()
    // ACT
    const { asFragment } = render(<MyButton alert={alertFunction} />);

    // ASSERT
    expect(asFragment()).toMatchSnapshot();
})

