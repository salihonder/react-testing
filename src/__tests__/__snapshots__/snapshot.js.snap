// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Snapshot test for Button 1`] = `
<DocumentFragment>
  <button>
    Some
  </button>
</DocumentFragment>
`;

exports[`Snapshot test for Header 1`] = `
<DocumentFragment>
  <h1
    style="color: red;"
  >
    Something
  </h1>
</DocumentFragment>
`;
